### PR TITLE
MB-14179-Delete-Line-Items-PPM

### PR DESCRIPTION
**Pull request by dymonmoore**
## Summary
This pull request connects the backend functionality from [MB-14903](https://dp3.atlassian.net/browse/MB-14903) to the PPM Shipment Review Page in order to delete line items such as (weight, expense, and pro-gear tickets). 

Solution Design for this PR: 
When the user clicks delete....

- A model will appear on the page displaying one of two options ('Yes, Delete' or 'No, Keep it')
- After a user says they want to delete a closeout item, we show a success or failure alert. 
- If a user deletes another item, the new success/failure alert will replace any existing success/failure alert. (only one success/error alert present)
- Success alert says Set/Trip/Receipt # successfully deleted.
- Error alert says Something went wrong deleting Set/Trip/Receipt #.  Please try again. 
- Page should refresh after deletion

## Verification Steps for Author and Reviewer

These are to be checked by the author and reviewer.

- [ ] Login to readyForFinalComplete@ppm.approved (milmove)
- [ ] Delete the 1st weight moved ticket
- [ ] Verify alert appears, is green and says 'Trip 1 successfully deleted.'
- [ ] Delete the 1st pro-gear ticket
- [ ] Verify alert appears, is green and says 'Set 1 successfully deleted.'
- [ ] Delete the 1st expense ticket
- [ ] Verify alert appears, is green and says 'Receipt 1 successfully deleted.'

[MB-14903]: https://dp3.atlassian.net/browse/MB-14903?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dymonmoore** [comment]
Hey Reviewers, any suggestions on how to improve the code quality on lines 104-134?

**Comment by dymonmoore** [comment]
Also, is the alert appearing too soon?

**Comment by mboren** [comment]
yeah there is a bug with the alert updating too early. when there's already an alert, the text will update when the delete modal is opened rather than when the thing is actually deleted. i think i know how to fix it

File: src/pages/MyMove/PPM/Closeout/Review/Review.jsx [hunk 1]
```diff
@@ -157,6 +199,7 @@ const Review = () => {
                 itemToDelete={itemToDelete}
               />
             )}
+            {alert && <Alert type={alert.type}> {setAlertMessage(alert.type, itemToDelete.itemNumber)} </Alert>}
```
Comment (mboren) [inline src/pages/MyMove/PPM/Closeout/Review/Review.jsx:202]: itemToDelete gets updated when the modal opens, so if an alert is already visible, and you open the delete modal, the alert message will change to the item you just clicked on, even though it hasnt been deleted yet

**Review by mandaem — COMMENTED** [review]
It deleted all 3 types for me. just needs some tests modified and It does look like you need a break line between the 2 alerts as shown in figma.  
Screenshot 2023-02-21 at 3 18 34 PM

File: src/pages/MyMove/PPM/Closeout/Review/Review.jsx [hunk 2]
```diff
@@ -93,18 +114,19 @@ const Review = () => {
...
-    }
+  const onDeleteSubmit = (itemType, itemId, itemNumber) => {
+    deleteLineItem(mtoShipment.ppmShipment.id, itemType, itemId)
+      .then(() => {
+        setIsDeleteModalVisible(false);
+        getMTOShipmentsForMove(mtoShipment.moveTaskOrderID).then((moveResponse) =>
+          dispatch(updateMTOShipment(moveResponse.mtoShipments[mtoShipment.id])),
+        );
+      })
+      .then(() => setAlert({ type: 'success', message: `${itemNumber} successfully deleted.` }))
```
Comment (duncan-truss) [inline src/pages/MyMove/PPM/Closeout/Review/Review.jsx:125]: I don't know too many other examples of chaining multiple `then()` callbacks.  I might have waited to set the alert until after the getMTOShipmentsForMove resolves and we call dispatch to Redux.  It seems to be alright the way it's working though.

File: src/pages/MyMove/PPM/Closeout/Review/Review.test.jsx [hunk 3]
```diff
@@ -419,14 +520,21 @@ describe('Review page', () => {
...
     const weightTicket = mockMTOShipmentWithWeightTicket.ppmShipment.weightTickets[0];
     await waitFor(() => {
-      expect(mockDeleteWeightTicket).toHaveBeenCalledWith(weightTicket.id, weightTicket.eTag);
+      expect(mockDeleteWeightTicket).toHaveBeenCalledWith(
+        mockMTOShipmentWithWeightTicket.ppmShipment.id,
+        weightTicket.id,
+      );
+    });
+    await waitFor(() => {
+      expect(screen.getByText('Trip 1 successfully deleted.'));
```
Comment (duncan-truss) [inline src/pages/MyMove/PPM/Closeout/Review/Review.test.jsx:529]: Just for other people who might be reading, we mocked the get shipments API response to exclude the item deleted. However, I'm not sure the dispatch call to redux would retrigger our selectMTOShipmentById hook to re-render and get that new value in a test scenario.  Might be best left for an E2E test and not a component unit test.

**Review by duncan-truss — APPROVED** [review]

**Review by dinabeigelman — APPROVED** [review]
lgtm!

**Pull request merged**